source/ScreensaverHero.hx

```diff
@@ -0,0 +1,92 @@
+package;
+
+import flixel.FlxG;
+import flixel.FlxSprite;
+import flixel.math.FlxRandom;
+import flixel.util.FlxSpriteUtil;
+
+class ScreensaverHero extends FlxSprite
+{
+	public var direction:Int;
+
+	public function new(x:Int, y:Int)
+	{
+		super(x, y);
+		var randomDirection:FlxRandom = new FlxRandom();
+		this.direction = randomDirection.int(0, 7);
+	}
+
+	override function update(elapsed:Float)
+	{
+		if (this.x < 500)
+		{
+			this.color = 0xff0000;
+		}
+		if (this.x > 500)
+		{
+			this.color = 0x00ff00;
+		}
+		if (this.direction == 0)
+		{
+			this.x--;
+		}
+		else if (this.direction == 1)
+		{
+			this.x--;
+			this.y--;
+		}
+		else if (this.direction == 2)
+		{
+			this.y--;
+		}
+		else if (this.direction == 3)
+		{
+			this.x++;
+		}
+		else if (this.direction == 4)
+		{
+			this.x++;
+			this.y++;
+		}
+		else if (this.direction == 5)
+		{
+			this.y++;
+		}
+		else if (this.direction == 6)
+		{
+			this.x--;
+			this.y++;
+		}
+		else
+		{
+			this.x++;
+			this.y--;
+		}
+		if (FlxG.mouse.pressed)
+		{
+			this.direction++;
+			if (this.direction == 8)
+			{
+				this.direction = 0;
+			}
+		}
+		if (this.x <= 0)
+		{
+			this.x = 1000;
```

Comment:
Let's also use `FlxG.width` here, like you're doing in the conditional later. 'magic numbers' like these aren't very programmer friendly, especially if you're trying to get someone new up to speed or remember what the code was supposed to do after some time away. 

It also adds overhead if you ever decide to change the dimensions of your game window, since you'd have to remember to come back to this spot and change these numbers to match the new width and height.

Same idea applies to the Y-axis wrapping below, I'll mark the spot with another comment.